### PR TITLE
Log a warning if multiple status contexts matched during aggregation

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/119

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

File: controller/status.go [hunk 1]
```diff
@@ -277,6 +263,11 @@ var (
 	}
 )
 
+// Aggregator provides methods for aggregating status.
+type Aggregator struct {
```
Comment (zeisss) [inline controller/status.go:267]: either make the functions on this type public as well or hide the type. A public type with no functions is useless.

**Comment by zeisss** [comment]
LGTM otherwise

File: controller/status.go (hunk 1 above)
Comment (JosephSalisbury) [inline controller/status.go:267]: https://github.com/giantswarm/inago/pull/133/files#diff-9155f40475e332d8aee80a437105af6bR279 AggregateStatus is public

File: controller/status.go [hunk 2]
```diff
@@ -285,29 +276,55 @@ var (
...
+	}
+
+	numAggregatedStatuses := len(aggregatedStatuses)
+
+	if numAggregatedStatuses == 0 {
+		return "", maskAnyf(invalidUnitStatusError, "fc: %s, fd: %s, sa: %s, ss: %s", fc, fd, sa, ss)
+	}
+
+	if numAggregatedStatuses > 1 {
+		a.Logger.Warning(nil, "Multiple statuses found for: fc: %s, fd: %s, sa: %s, ss: %s", fc, fd, sa, ss)
```
Comment (xh3b4sd) [inline controller/status.go:306]: Do we know when this happens? Neither the issue nor the PR provide some helpful information.

Comment (JosephSalisbury) [inline controller/status.go:306]: This happened and was fixed in https://github.com/giantswarm/inago/pull/116 fixed it - the warning has been kept in the case that it happens again.

**Pull request merged**